addons/nodot/interaction/Teleport3D.gd

```diff
@@ -0,0 +1,101 @@
+##  Teleport3D is an Area3D which, when touched, teleports the collider to another location either instantly or after a defined time
+class_name Teleport3D extends Area3D
+
+@export_category("Nodes")
+##  Target node for copy the coords
+@export_node_path("Node3D") var target
+##  Reference a node to teleport by target
+@export_node_path("CharacterBody3D") var player
+
+@export_category("Others Variables")
+##  Size of collision area
+@export var size = Vector3(3.372, 3.821, 1)
+##  Time to teleport (in seconds, and this variable is optional)
+@export var time_teleport : int = 0
+
+##  Get position of "Player" 
+@onready var node_player = get_node(player)
+##  Get position of "Target"
+@onready var position_target = get_node(target)
+
+##  Collision Shape
+var node_collision = CollisionShape3D.new()
+##  Timer node
+var node_time = Timer.new()
+
+##  Check variable if "Player" is in "area"
+var can_teleport_area = false
+##  Check variable, if "Player" wait a time
+var can_teleport_time = false
+
+func _ready():
+	#  add collision to scene
+	node_collision.name = "CollisioArea"
+	
+	#  Procces to select automatic or manully a shape
+	var shape
+	var check_num_collision = 0
+	
+	#  Take first node and copy the shape, erase anothers nodes (automatic).
+	for child in get_children():
+		if child is CollisionShape3D:
+			check_num_collision += 1
+			if check_num_collision == 1:
+				##  check if "child" have "shape"
+				if child.shape != null:
+					child.name = "CollisioArea"
+					shape = child.shape
+					node_collision.shape = shape
+			else:
+				child.free()
+	
+	#  check if shape have a form, if not have, create one (manully)
+	if shape == null:
+		shape = BoxShape3D.new()
+		#  Add size to collision and add the shape to "CollisioArea"
+		shape.size = size
+		node_collision.shape = shape
+		
+		#  check if in the scene exist another collisionshape
+		for child in get_children():
+			if child is CollisionShape3D:
+				child.free()
```

Comment:
Always use `queue_free` as it's safer to ensure the var continues to be available in the current frame.